intelmq/bots/experts/sieve/README.md

```diff
@@ -0,0 +1,166 @@
+# Sieve Bot
+
+The sieve bot is used to filter and/or modify events based on a set of rules. The
+rules are specified in an external configuration file and with a syntax similar
+to the [Sieve language](http://sieve.info/) used for mail filtering.
+
+Each rule defines a set of matching conditions on received events. Events can be
+matched based on keys and values in the event. If the processed event matches a
+rule's conditions, the corresponding actions are performed. Actions can specify
+whether the event should be kept or dropped in the pipeline (filtering actions)
+or if keys and values should be changed (modification actions).
+
```

Comment:
Would it be possible to specify a specific outgoing pipeline as an action?

Example: 
```
  if $condition
  then
    send-to "pipeline-A";
  fi
```

Reply:
This is not supported by the core.

Reply:
In our instance, we are already (ab)using a newly introduced key called `keep` (boolean), which can be set anywhere in the pipeline. Downstream, events with `keep==False` are dropped.

This can be generalized by introducing a new key, e.g. `route`, with values that describe a specific destination. But this requires to put filter bots upstream of every destination such that only the corresponding events are kept.

If a routing model was provided in the core, we would certainly have use-cases for it.

Reply:
I wonder why no issue existed for this which and created #1088 for it